app/services/activitypub/prepare_followers_synchronization_service.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class ActivityPub::PrepareFollowersSynchronizationService < BaseService
+  include JsonLdHelper
+
+  def call(account, params)
+    @account = account
+
+    return unless params['collectionId'] == @account.followers_url
+    return if invalid_origin?(params['url'])
+    return if @account.followers_hash('local') == params['digest']
```

Comment:
Could be part of the guard clause above. This whole method might look better with just one guard clause, they are simple comparisons.

If the goal of supporting both `nil` and `"local"` arguments in `followers_hash` is code readability, you could create a method like `local_followers_hash` that calls `followers_hash` with `nil`.